x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -99,7 +99,16 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
   }
 
   public async find<T = unknown>(options: SavedObjectsFindOptions) {
-    await this.ensureAuthorized(options.type, 'find', options.namespace, { options });
+    if (
+      this.getSpacesService() == null &&
+      Array.isArray(options.namespaces) &&
+      options.namespaces.length > 0
+    ) {
+      throw this.errors.createBadRequestError(
+        `_find across namespaces is not permitted when the Spaces plugin is disabled.`
+      );
+    }
+    await this.ensureAuthorized(options.type, 'find', options.namespaces, { options });
```

Comment:
My intent here was to prevent throwing an [existing authorization error](https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/x-pack/plugins/security/server/authorization/check_saved_objects_privileges.ts#L33-L36) when multiple namespaces are specified when Spaces is disabled. This essentially gives a more helpful error message than what would have been thrown.

I'm not opposed to this change though. Maybe we could change the logic within [`checkSavedObjectsPrivilegesWithRequest`](https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/x-pack/plugins/security/server/authorization/check_saved_objects_privileges.ts#L24) so that we always check privileges "globally" when spaces is disabled, rather than throwing an error if a multi-namespace check is requested when spaces is disabled?

Reply:
I've rewritten this logic. The new flow:

1) If Spaces is disabled, then always check privileges globally. No need to check against the specific namespaces in this scenario. If they're authorized globally, then it doesn't matter what the namespaces are; If they aren't authorized globally, then they wouldn't be authorized to use these APIs anyway.
2) If Spaces is enabled, then check privileges against the provided namespace(s) as-is